### PR TITLE
Use provider eksctl method to run eksctl because the binary is not in…

### PR DESCRIPTION
**Pull request by ycliuhw**
### Checklist

 - [ ] ~Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?~
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [ ] ~Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?~
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

Use EKS provider's eksctl method to run eksctl command because the binary was put to juju home but not in PATH;

### CODE REVIEW

File: acceptancetests/jujupy/k8s_provider/base.py [hunk 1]
```diff
@@ -211,13 +211,14 @@ def patch_configmap(self, namespace, cm_name, key, value):
         cm['data'] = data
         self.kubectl_apply(json.dumps(cm))
 
-    def sh(self, *args):
+    def sh(self, *args, shell=False):
         args = [str(arg) for arg in args]
```
Comment (howbazaar) [inline acceptancetests/jujupy/k8s_provider/base.py:215]: If we are going to specify shell=True, we need to quote the args. See https://docs.python.org/3/library/subprocess.html#security-considerations

Comment (ycliuhw) [reply acceptancetests/jujupy/k8s_provider/base.py:215]: ah, right. fixed

File: acceptancetests/jujupy/k8s_provider/base.py [hunk 2]
```diff
@@ -211,13 +212,14 @@ def patch_configmap(self, namespace, cm_name, key, value):
         cm['data'] = data
         self.kubectl_apply(json.dumps(cm))
 
-    def sh(self, *args):
-        args = [str(arg) for arg in args]
+    def sh(self, *args, shell=False):
+        args = [quote(str(arg)) for arg in args]
```
Comment (howbazaar) [inline acceptancetests/jujupy/k8s_provider/base.py:216]: We only want to quote the args if shell is True, right?

Comment (ycliuhw) [reply acceptancetests/jujupy/k8s_provider/base.py]: ah, right. ty!

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**